components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/indexing/indexer/CustomAPIIndexerTest.java

```diff
@@ -67,9 +71,17 @@ public void init() throws RegistryException {
      * @throws RegistryException Registry Exception.
      */
     @Test
-    public void testIndexDocumentForMigratedAPI() throws RegistryException {
+    public void testIndexDocumentForMigratedAPI() throws Exception {
```

Comment:
Do not throw generic Exception. Throw specific exceptions comma separated.

Reply:
This is fine for a test case I think

Reply:
@malinthaprasan When a test get failed due to an exception, generic Exception stacktrace will be logged. Then figuring out the exact cause would be difficult. What is the difference between test cases and non-test cases you mean in this case?

Reply:
This has been fixed. Anyway there is no difference in throwing as an "Exception" or "APIManagementException, RegistryException" here in method signature as we cast the exception to the corresponding throwing type and that does not change the content of the exception.